ledger/participant-integration-api/src/main/scala/platform/InMemoryState.scala

```diff
@@ -42,18 +43,24 @@ private[platform] class InMemoryState(
     // TODO LLP: Reset the in-memory state only if the initialization ledgerEnd
     //           is different than the ledgerEndCache.
     for {
+      // First stop the active dispatcher (if exists) to ensure
+      // that Ledger API subscriptions racing with `initializeTo`
+      // do not observe an inconsistent state.
+      _ <- dispatcherState.stopDispatcher()
```

Comment:
This is not consistent with the comment, since the comment state no inconsistent state will be observed, but yet this future can finish (because of the timeout), before all the Dispatcher streams complete.
If it is not a problem that all the not-yet-terminated streams are observing changing state, I would suggest not waiting for this Future
If it is a problem, I would suggest waiting for completion for all the streams (this is also probably faster if we terminate them with an error? maybe we could do that also in this PR (and resolve the related TODO too)?)

Reply:
a **new** LAPI subscription racing with `initializedTo` will end in a failure. Existing ones are indeed not guarded.

Reply:
My comment is not about new subscriptions, but old ones which before terminating do something (after the timeout), based on potentially inconsistent state. Do you think that is a problem?

Reply:
Solved the other TODO in this PR which should ensure timely termination of all existing subscriptions with failure. In the current state, nothing bad should happen if an active stream experiences this transition state (it will not get new data anyway as its dispatcher is shutdown). But as a safeguard for future, I'm removing the shutdown timeout for this step (as it should be eager anyway).

Reply:
:+1: